drf-api/schedule/tasks.py

```diff
@@ -0,0 +1,60 @@
+import csv
+from io import StringIO
+
+from django.contrib.auth.models import User
+from django.core.mail import EmailMessage
+from marked.celery import app
+
+from schedule.serializers import ProviderSerializer
+
+
+def write_file(writer):
+    providers = User.objects.all()
+    serializer = ProviderSerializer(providers, many=True)
+
+    for provider in serializer.data:
+        schedulings = provider["scheduling"]
+        for scheduling in schedulings:
+            writer.writerow(
+                [
+                    scheduling["provider"],
+                    scheduling["client_name"],
+                    scheduling["client_email"],
+                    scheduling["client_phone"],
+                    scheduling["confirmed"],
+                    scheduling["states"],
+                ]
+            )
+
+
+def generate_file(output):
+    writer = csv.writer(output)
+    writer.writerow(
+        [
+            "provider",
+            "client_name",
+            "client_email",
+            "client_phone",
+            "confirmed",
+            "states",
+        ]
+    )
+    write_file(writer)
+
+
+@app.task
+def generate_report_provider():
+    output = StringIO()
+    generate_file(output)
+
+    email = EmailMessage(
+        subject="Marked -> Provider report",
+        body="Em anexo relatório solicitado.",
+        from_email="ernesto.terra2003@gmail.com",
+        to=["ernesto.terra2003@gmail.com"],
```

Comment:
Aqui podemos usar uma variável de ambiente para o `from_email`, só pra evitar expôr seu e-mail no GitHub 😅 

E o `to` pode ser o próprio usuário que solicitou o relatório. Para isso, você vai precisar passar alguma informação quando chamar a task. Por exemplo: `generate_report.provider.delay(user_id=request.user.id)`